clients/src/main/java/org/apache/kafka/common/metrics/stats/Rate.java

```diff
@@ -34,6 +34,7 @@ public class Rate implements MeasurableStat {
 
     protected final TimeUnit unit;
     protected final SampledStat stat;
+    protected final long window;
```

Comment:
// The metric config which holds custom time window configuration.
```suggestion
    protected final MetricConfig overriddenMetricConfig;
```